src/H5Dchunk.c

```diff
@@ -5603,156 +5607,73 @@ H5D__chunk_collective_fill(const H5D_t *dset, H5D_chunk_coll_fill_info_t *chunk_
 
     /* Check if we have any chunks to write on this rank */
     if (num_blocks > 0 || (leftover && leftover > mpi_rank)) {
```

Comment:
Minor nitpick with pre-existing code - the first leftover is redundant since mpi_rank will never be less than 0. This could be `if (num_blocks > 0 || leftover > mpi_rank) {`

Reply:
Done with this and subsequent comments.